libraries/encrypted-db/src/main/kotlin/io/element/encrypteddb/EncryptedDatabase.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.encrypteddb
+
+import android.content.Context
+import com.squareup.sqldelight.android.AndroidSqliteDriver
+import com.squareup.sqldelight.db.SqlDriver
+import io.element.encrypteddb.passphrase.PassphraseProvider
+import net.sqlcipher.database.SupportFactory
+
+/**
+ * Creates an encrypted version of the [SqlDriver] using SQLCipher's [SupportFactory].
+ * @param passphraseProvider Provides the passphrase needed to use the SQLite database with SQLCipher.
+ */
+class SqlCipherDriverFactory(
+    private val passphraseProvider: PassphraseProvider,
```

Comment:
My first idea was trying to pass this using DI, but as there might be several instances of this in the future and its configuration is not trivial, maybe it would be better to just pass a `passphrase: String` parameter retrieved in the `provides...` method that creates this driver instead.